### PR TITLE
Issue #22 - BIO.wander

### LINKED ISSUE #22: Implement BIO.wander from Monix Task
Please put the method in other companion objects as well (`Task`, `UIO`) with appropriate return type if it applies

General flow:
- Comment in this issue to let others know that someone is working on it
- Copy scaladoc and the implementation from [Monix Task](https://github.com/monix/monix) in similar place
- Modify signature to take advantage of typed errors if possible
- Implementation might require some changes, in most cases the compiler should guide you, also look for already ported operators for inspiration. The most common change is to use `BiCallback` instead of `Callback` to be able to signal errors on two different channels ("typed" channel and hidden channel for "fatal" errors).
- Copy the corresponding tests from Monix. Try to use `BIO` instead of `Task` with `E` different than `Throwable` when possible to uncover any possible bugs in the internals. 
- If you test for a failure, try to test for both "expected" failure (error from `E`, it will be `CancelableFuture(Left))` after `runToFuture`) and "fatal" failure (it will be a failed `CancelableFuture`).
- Take a look at the existing tests to see if there aren't any commented ones using newly implemented function and uncomment them.

Don't be shy to ping me if you need any advice or want to consult the solution. I really appreciate the involvement and I'm happy to help!

### PR DESCRIPTION
**Pull request by pk044**
Closes #22 

I commented one test since BIO.memoize is not implemented yet - so I think we should wait with merging this PR.

### CODE REVIEW

File: core/shared/src/test/scala/monix/bio/TaskWanderSuite.scala [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+  test("Task.wander should wrap exceptions in the function") { implicit s =>
+    val ex = DummyException("dummy")
+    val task1 = BIO.wander(Seq(0)) { i =>
+      throw ex
+      BIO.now(i)
+    }
+
+    val result1 = task1.runToFuture; s.tick()
+    assertEquals(result1.value, Some(Success(Left((ex)))))
```
Comment (Avasil) [inline core/shared/src/test/scala/monix/bio/TaskWanderSuite.scala]: This one is wrong, I made a PR to fix it #57

`task1` has `E = Nothing` and yet it returns a `Left(ex)` 😅

File: core/shared/src/test/scala/monix/bio/TaskWanderSuite.scala [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  test("BIO.wander should be stack safe for synchronous tasks") { implicit s =>
+    val count = if (Platform.isJVM) 200000 else 5000
+    val seq = for (i <- 0 until count) yield 1
+    val composite = BIO.wander(seq)(BIO.now).map(_.sum)
+    val result = composite.runToFuture
+    s.tick()
+    assertEquals(result.value, Some(Success(Right(count))))
+  }
+
+  // test("Task.wander runAsync multiple times") { implicit s =>
```
Comment (Avasil) [inline core/shared/src/test/scala/monix/bio/TaskWanderSuite.scala]: It's fine to leave it commented, there are lots of tests like that until we have everything

File: core/shared/src/test/scala/monix/bio/TaskWanderSuite.scala (hunk 1 above)
Comment (pk044) [reply core/shared/src/test/scala/monix/bio/TaskWanderSuite.scala]: ok, I'll fix that once the #57 is merged :)

**Comment by pk044** [comment]
@Avasil I adjusted the PR to the changes introduced by #57

**Pull request merged**